### DISCUSSION TITLE
using chrome CDP

### DISCUSSION THREAD

**Discussion by jansengorilla** [stashapp/stash#4991] [Q&A] [answered]
Using docker. Having installed browserless/chromium, i get the following error, having set 'driver / useCDP: true' in the yml file of the scraper.

failed to modify wsURL: invalid character '<' looking for beginning of value 

What am i missing? 

Many thanks!

**Comment by feederbox826** [comment, marked as answer]
browserless-chromium is not CDP or a CDP equivalent.
You need to use https://hub.docker.com/r/chromedp/headless-shell/ instead.

**Reply by cytec** [reply]
Well yes and no.
browserless offers the same features as CDP + some extras and they do support `chromedp` which is used by stash see https://docs.browserless.io/libraries/chromedp

imho the main reason why browserless doesn't work is that stash allows chromedp to modify the url here https://github.com/stashapp/stash/blob/0621d87133a9e810687e0da27fd3ea65095873a0/pkg/scraper/url.go#L149 which could be "fixed" by adding `chromedp.NoModifyURL` 

a simple way to support both would be to add the `chromedp.NoModifyURL`  flag if the url was given with `ws://` or `wss://` (or maybe make it an option to disable url modification)

also: the [isCDPathWS](https://github.com/stashapp/stash/blob/develop/pkg/scraper/cache.go#L49) function doesn't check `wss://`

BTW: when you put https://hub.docker.com/r/chromedp/headless-shell/ behind a reverse proxy (to get rid of the port, add basic auth or whatever) chromedp fails with `failed to modify wsURL: address XXXX missing port in address` 

i've got a remote instance of browserless running and it would be nice to be able to use it. spinning up another container that does the same but is completely open doesn't seem very appealing to me tbh
